src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -89,7 +92,47 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// <summary>
         /// Additional information attached to a completion item by it creator.
         /// </summary>
-        public ImmutableDictionary<string, string> Properties { get; }
+        public ImmutableDictionary<string, string> Properties
+        {
+            get
+            {
+                if (_propertiesAsImmutableDictionary is null)
+                    _propertiesAsImmutableDictionary = _properties.ToImmutableDictionary();
+
+                return _propertiesAsImmutableDictionary;
+            }
+        }
+
+        internal IEnumerable<KeyValuePair<string, string>> GetProperties()
```

Comment:
This always just returns the immutable array

```suggestion
        internal ImmutableArray<KeyValuePair<string, string>> GetProperties()
```

Reply:
That was intentional, as I felt a bit burned by the Properties getter returning a concrete type. However, maybe ImmutableArray is fine to return. I'll switch it.

Reply:
being internal, it's fine.  if it's the public surface area, then it can warrant more scrutiny.